akka-http-core/src/test/scala/akka/http/impl/util/StreamUtilsSpec.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2003-2016 Monitise Group Limited. All Rights Reserved.
+ *
+ * Save to the extent permitted by law, you may not use, copy, modify,
+ * distribute or create derivative works of this material or any part
+ * of it without the prior written consent of Monitise Group Limited.
+ * Any reproduction of this material must contain this notice.
+ */
+package akka.http.impl.util
+
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, Source}
+import akka.testkit.AkkaSpec
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.Failure
+
+class StreamUtilsSpec extends AkkaSpec {
+  implicit val materializer = ActorMaterializer()
+
+  "captureTermination" should {
+    "signal completion" when {
+      "upstream terminates" in {
+        val (newSource, whenCompleted) = StreamUtils.captureTermination(Source(List(1, 2, 3)))
+
+        newSource.runWith(Sink.ignore)
+
+        Await.result(whenCompleted, 100.millis) shouldBe ()
```

Comment:
`100.millis` is way to short for some CI environments, please use `remainOrDefault` (or at least 3.seconds if it's not important to fail fast)